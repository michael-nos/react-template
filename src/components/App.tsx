export const App = () => {
  return (
    <>
      <h1>Hey there, time to make a fast app!</h1>
    </>
  );
};
