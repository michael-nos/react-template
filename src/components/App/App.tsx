import { Route, Routes } from 'react-router-dom';
import { NotFound } from '@/components/NotFound';
import { Home } from '@/routes/Home';
import { Test } from '@/routes/Test';

export const App = () => {
  return (
    <Routes>
      <Route path='/' element={<Home />} />
      <Route path='/test' element={<Test />} />
      <Route path='*' element={<NotFound />} />
    </Routes>
  );
};
