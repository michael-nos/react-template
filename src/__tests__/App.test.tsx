import { render, screen } from "@testing-library/react";
import { App } from "@/components/App";

describe("App", () => {
	it("should match the snapshot", () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});

	it("should render the heading text", () => {
		render(<App />);
		expect(screen.getByText(/Hey there, time to make a fast app!/i)).toBeInTheDocument();
	});
});
