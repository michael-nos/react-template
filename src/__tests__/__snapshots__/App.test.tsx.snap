// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match the snapshot 1`] = `
<div>
  <div>
    Mock App Component
  </div>
</div>
`;
